milli/src/update/index_documents/transform.rs

```diff
@@ -92,7 +96,7 @@ impl Transform<'_, '_> {
         // We extract the primary key from the first document in
         // the batch if it hasn't already been defined in the index
         let first = documents.peek().and_then(|r| r.as_ref().ok());
```

Comment:
I found this line to make the engine return an error that is not the right one, when there is a decoding error we simply ignore it (`.ok()`), and then this a "missing primary key" error that is thrown due to the fact that no primary key was guessed **and** that the `alternate_name` is None because the first document is invalid.

We should early return when we find that the first document is erroneous, it would be better, but the error doesn't impl `Clone` so we can't return it from the `peek`. We should check that the peeked document is `Err` and if it is we should next and return the error (`unwrap_err`).

Reply:
```suggestion
        let first = match iter.peek() {
            Some(Err(_)) => iter.next().transpose()?,
            Some(Ok(x)) => Some(x),
            None => None,
        };
```

Untested in this context

Reply:
this doesn't compile, but I wrote something similar.